algo/src/main/java/com/hartwig/serve/extraction/gene/GeneLevelExtractor.java

```diff
@@ -60,6 +59,10 @@ public GeneAnnotation extract(@NotNull String gene, @NotNull EventType type, @No
             return extractGeneLevelEvent(gene, event);
         } else if (type == EventType.PROMISCUOUS_FUSION && fusionGeneChecker.isValidGene(gene)) {
             return extractPromiscuousFusion(gene);
+        } else if (type == EventType.NEGATIVE && exomeGeneChecker.isValidGene(gene)) {
+            return extractNegativePositiveEvent(gene, GeneEvent.NEGATIVE, type);
+        } else if (type == EventType.POSITIVE && exomeGeneChecker.isValidGene(gene)) {
+            return extractNegativePositiveEvent(gene, GeneEvent.POSITIVE, type);
```

Comment:
I am not sure if I understood you correctly? doe you mean this: "if (type instanceof EventType) {}", this give a warning because this statement is always true. So think you mean something different!

Also, I make a switch statement for the different types.

Reply:
See my commit of 21st of May!